docs/authentication.md

```diff
@@ -0,0 +1,145 @@
+  * The token is stored on server side (in `authtoken_token` table), alongside a creation date and and the user ID.
+  * The token is a random string encoded in hexadecimal format. [Source](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/models.py)
+  * The token should be included in the `Authorization` HTTP header
+- The API has a custom layer to handle the token expiration based on its creation date. [Source](../libs/expiry_token_authentication.py)
+
+1. Post the credentials using JSON data to download the authentication token.
+
+```bash
+curl \
+  --request POST \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/json' \
+  --data '{"username":"org-1","password":"p@sswr0d44"}' \
+  http://substra-backend.org-1.com/api-token-auth/
+{"token":"<auth_token_value>","expires_at":"59841.718064"}
+```
+
+2. Fetch any asset with the authentication token in the header.
+
+```bash
+curl \
+  --header 'Accept: application/json' \
+  --header 'Authorization: Token <auth_token_value>' \
+  http://substra-backend.org-1.com/algo/
+{"count":0,"next":null,"previous":null,"results":[]}
+```
+
+## JWT cookie
+
+- URLs: `/me/login/`, `/me/refresh/`
+- Implemented in `users.views.authentication` [module](../backend/users/views/authentication.py)
+- Based on `rest_framework_simplejwt` [library](https://django-rest-framework-simplejwt.readthedocs.io/en/latest/index.html).
+  * Generate a [JWT token](https://jwt.io/introduction)
+    - Header: contains the signing algorithm, encoded in Base 64.
+    - Payload: contains authentication data (user ID, expiration date) encoded in Base 64.
+    - Signature: the header and the payload encrypted with a secret.
+- The API has a custom layer to split the JWT token in 3 cookies.
+  * A long lifespan cookie which contains the header and the payload.
+  * A short lifespan cookie which contains the signature, using HTTP only (not accessible by JS code to prevent from Cross-Site-Scripting / XSS attacks).
+  * A short lifespan cookie which contains the refreshing expiration date, using HTTP only.
+
+1. Post the credentials using JSON data to download the cookies.
+
+```bash
+curl \
+  --request POST \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/json' \
+  --data '{"username":"org-1","password":"p@sswr0d44"}' \
+  --cookie-jar cookie-jwt.jar \
+  http://substra-backend.org-1.com/me/login/
+{"token_type":"access","exp":1665061851,"iat":1664975451,"jti":"8f6d98b667024db59dcc55f923db8d22","user_id":1}
+
+cat cookie-jwt.jar
+#HttpOnly_.org-1.com    TRUE    /    FALSE    1665580790    refresh           <refresh_value>
+#HttpOnly_.org-1.com    TRUE    /    FALSE    0             signature         <signature_value>
+.org-1.com              TRUE    /    FALSE    1665062390    header.payload    <header_payload_value>
```

Comment:
A `#` is missing here

Reply:
No this raw output. Curl add `#` for `HttpOnly`